includes/database/queries/class-edd-db-column.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/**
+ * Orders: EDD_DB_Query class
+ *
+ * @package Plugins/EDD/Database/Queries
+ */
+
+// Exit if accessed directly
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Base class used for querying custom database tables.
+ *
+ * @since 3.0.0
+ *
+ * @see EDD_DB_Query::__construct() for accepted arguments.
+ */
+class EDD_DB_Column {
+
+	/**
+	 * Name for the database column
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 * @var string
+	 */
+	public $name = '';
+
+	/**
+	 * Type of database column
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 * @var string
+	 */
+	public $type = '';
+
```

Comment:
What's the reason for these properties being `public`? Can we make them `protected` or `private` and use a method to retrieve them instead?

Reply:
The reason is it's easier for me to architect these bits fully public, and then protect or privatize them later once it's all working.

Protected is probably best IMO. I think we can add methods to get private vars once we start enforcing the shape/sanitization/validation of them.

Reply:
Just tested setting these as `protected` and remembered that even with magic getters, WordPress's list utility functions and methods use `array_key_exists()` internally, so these will need to either:

* Remain explicitly public
* Not be defined at all, which will default them to public when they are set

I think being explicit here is better, even if it's exposing these vars a bit.